engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -18,46 +20,48 @@ object EngineJobExecutionActor {
 
   /** Commands */
   sealed trait EngineJobExecutionActorCommand
-  case class Execute(jobKey: BackendJobDescriptorKey) extends EngineJobExecutionActorCommand
+  case object Execute extends EngineJobExecutionActorCommand
 
-  case class JobRunning(jobDescriptor: BackendJobDescriptor, backendJobExecutionActor: ActorRef)
+  final case class JobRunning(jobDescriptor: BackendJobDescriptor, backendJobExecutionActor: ActorRef)
 
-  def props(executionData: WorkflowExecutionActorData, factory: BackendLifecycleActorFactory,
+  def props(jobDescriptorKey: BackendJobDescriptorKey, executionData: WorkflowExecutionActorData, factory: BackendLifecycleActorFactory,
             initializationData: Option[BackendInitializationData], restarting: Boolean) = {
-    Props(new EngineJobExecutionActor(executionData, factory, initializationData, restarting))
+    Props(new EngineJobExecutionActor(jobDescriptorKey, executionData, factory, initializationData, restarting))
   }
 }
 
-class EngineJobExecutionActor(executionData: WorkflowExecutionActorData,
+class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
+                              executionData: WorkflowExecutionActorData,
                               factory: BackendLifecycleActorFactory,
                               initializationData: Option[BackendInitializationData],
-                              restarting: Boolean) extends LoggingFSM[EngineJobExecutionActorState, Unit] with WorkflowLogging {
+                              restarting: Boolean) extends LoggingFSM[EngineJobExecutionActorState, Unit] with WorkflowLogging with ServiceRegistryClient {
 
   override val workflowId = executionData.workflowDescriptor.id
 
   startWith(Pending, ())
 
   when(Pending) {
-    case Event(Execute(jobKey), _) =>
+    case Event(Execute, _) =>
       if (restarting) {
-        context.actorOf(JobStoreReader.props()) ! QueryJobCompletion(jobKey)
+        val jobStoreKey = jobKey.toJobStoreKey(workflowId)
+        serviceRegistryActor ! QueryJobCompletion(jobStoreKey)
         goto(CheckingJobStatus)
       } else {
         prepareJob(jobKey)
       }
   }
 
   when(CheckingJobStatus) {
-    case Event(JobNotComplete(jobKey), _) =>
+    case Event(JobNotComplete(_), _) =>
```

Comment:
If we aren't going to verify we got the right jobKey here, we probably don't need it in the `JobNotComplete` message